multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +297,38 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounces() error {
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	for _, m := range memberClusterAnnounceList.Items {
+		var needToDelete bool
+		memberClusterAnnounce := m
```

Comment:
I do not get why you need this line. If you like to use "memberClusterAnnounce" rather than "m", you can just change "m" to "memberClusterAnnounce" in line 306.

Reply:
use memberClusterAnnounce now.

Reply:
@hjiajing I recalled that you added this line because of a golang ci failure, please run `make golangci` to double check it locally first.

Reply:
Added the temporary variable back. And removed the next "else" block because there's a "continue" above it.

Reply:
Could I know what failure we get if we remove `memberClusterAnnounce := m`?

Reply:
The error will be like 'Implicit memory aliasing in for loop'. This is also one of common mistake in golang, and the wiki provide the solution like @hjiajing  did here. https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable

Reply:
Ok. That makes sense to me now. Thanks for clarification!